src/3rd_party_repos.c

```diff
@@ -140,11 +142,44 @@ int third_party_add_repo(const char *repo_name, const char *repo_url)
 	return ret;
 }
 
-void third_party_set_repo(const char *state_dir, const char *path_prefix, struct repo *repo)
+static int third_party_create_state_dirs(const char *state_dir, const char *repo_name)
+{
+	int ret = 0;
+	char *dir;
+
+	/* check for existence */
+	string_or_die(&dir, "%s/3rd_party/%s", state_dir, repo_name);
```

Comment:
Why don't we just cal create_state_dirs() here?

Reply:
well I wasn't just calling create_state_dirs because it was creating an extra directory we don't really need in the state dir of 3rd_party repos. But I guess it really doesn't matter if that extra directory gets created, we just won't use it. I'll update this so we use create_state_dirs instead.